kvbc/tools/db_restore/db_restore.cpp

```diff
@@ -0,0 +1,122 @@
+// Concord
+//
+// Copyright (c) 2020-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "db_restore.hpp"
+#include "string.hpp"
+#include "merkle_tree_storage_factory.h"
+#include "categorization/base_types.h"
+#include "categorization/db_categories.h"
+
+namespace concord::kvbc::tools {
+using namespace std::placeholders;
+
+DBRestore::DBRestore() {
+  // clang-format off
+  cli_mandatory_options_.add_options()
+      ("db-path",
+      po::value<fs::path>()->required()->notifier(std::bind(std::mem_fn(&DBRestore::initRocksDB), this, _1)),
+      "rocksDB directory path");
+  cli_options_.add_options()("help,h", "produce help message");
+  cli_options_.add(integrity_check_.getOptions()).add(cli_mandatory_options_);
+  // clang-format on
+}
+void DBRestore::parseCLIArgs(int argc, char** argv) {
+  auto usage = [this]() { LOG_ERROR(logger_, cli_options_); };
+
+  // parse arguments
+  po::store(po::parse_command_line(argc, argv, cli_options_), var_map_);
+
+  // check arguments
+  try {
+    po::notify(var_map_);
+  } catch (const std::exception& e) {
+    usage();
+    throw;
+  }
+  if (var_map_.count("help")) {
+    usage();
+    std::exit(1);
+  }
+}
+
+void DBRestore::initRocksDB(const fs::path& rocksdb_path) {
+  LOG_DEBUG(logger_, rocksdb_path);
+  rocksdb_dataset_ = std::make_unique<v2MerkleTree::RocksDBStorageFactory>(rocksdb_path)->newDatabaseSet();
+  const auto linkStChain = true;
+  auto kvbc_categories = std::map<std::string, categorization::CATEGORY_TYPE>{
```

Comment:
Shouldn't this map be initialized once?

Reply:
I believe it's initialized in `categorization::KeyValueBlockchain()`

Reply:
It's initialized here, I was asking whether it should have been made static so that it is initialized only once rather than every time the function is called.

Reply:
it's called once in a run

Reply:
DBRestore is not a singleton, so the "called once" is not enforced, but I guess It's not that important.